packages/wrangler/src/metrics/metrics-dispatcher.ts

```diff
@@ -112,13 +135,49 @@ export function getMetricsDispatcher(options: MetricsConfigOptions) {
 			mode: "cors",
 			keepalive: true,
 			body: JSON.stringify(body),
-		}).catch((e) => {
-			logger.debug(
-				"Metrics dispatcher: Failed to send request:",
-				(e as Error).message
+		})
+			.then(() => {})
+			.catch((e) => {
+				logger.debug(
+					"Metrics dispatcher: Failed to send request:",
+					(e as Error).message
+				);
+			});
+
+		requests.push(request);
+	}
+
+	function printMetricsBanner() {
+		const metricsConfig = readMetricsConfig();
+		const lastShown = metricsConfig.permission?.bannerLastShown;
+		if (
+			metricsConfig.permission?.enabled &&
+			(!lastShown || isNewVersion(lastShown, wranglerVersion))
+		) {
+			logger.log(
+				chalk.gray(
+					`\nCloudflare collects anonymous telemetry about your usage of Wrangler. Learn more at https://github.com/cloudflare/workers-sdk/tree/main/telemetry.md`
+				)
 			);
-		});
+			metricsConfig.permission.bannerLastShown = wranglerVersion;
+			writeMetricsConfig(metricsConfig);
+		}
 	}
 }
 
 export type Properties = Record<string, unknown>;
+
+const isNewVersion = (stored: string, current: string) => {
+	const storedVersion = stored.split(".");
+	const currentVersion = current.split(".");
+	for (let i = 0; i < storedVersion.length; i++) {
+		const storedSegment = parseInt(storedVersion[i]);
+		const currentSegment = parseInt(currentVersion[i]);
+		if (currentSegment > storedSegment) {
+			return true;
+		} else if (currentSegment < storedSegment) {
+			return false;
+		}
+	}
+	return false;
+};
```

Comment:
This might not work with pre-release version (e.g. `0.0.0-76d43d8b1`). What if we don't care about whether it is a newer version but showing the banner whenever the version is changed?

Reply:
[57eca6c](https://github.com/cloudflare/workers-sdk/pull/7302/commits/57eca6ccfd98cfe174537d2ec59ff1195f2af952)